components/brave_wallet/browser/zcash/zcash_discover_next_unused_zcash_address_task.cc

```diff
@@ -14,18 +17,25 @@ namespace brave_wallet {
 
 ZCashDiscoverNextUnusedZCashAddressTask::
     ZCashDiscoverNextUnusedZCashAddressTask(
+        base::PassKey<class ZCashWalletService> pass_key,
         base::WeakPtr<ZCashWalletService> zcash_wallet_service,
-        mojom::AccountIdPtr account_id,
-        mojom::ZCashAddressPtr start_address,
+        const mojom::AccountIdPtr& account_id,
+        const mojom::ZCashAddressPtr& start_address,
         ZCashWalletService::DiscoverNextUnusedAddressCallback callback)
     : zcash_wallet_service_(std::move(zcash_wallet_service)),
-      account_id_(std::move(account_id)),
-      start_address_(std::move(start_address)),
+      account_id_(account_id.Clone()),
+      start_address_(start_address.Clone()),
       callback_(std::move(callback)) {}
 
 ZCashDiscoverNextUnusedZCashAddressTask::
     ~ZCashDiscoverNextUnusedZCashAddressTask() = default;
 
+void ZCashDiscoverNextUnusedZCashAddressTask::Start() {
+  CHECK(!started_);
```

Comment:
Same question as above - Do we have another option other than crashing that we can do here? We can consolidate conversation about it into the comment above and whatever we decide apply here as well.

Reply:
Single task must not be started several times - otherwise this is a bug. We can use DCHECK to prevent crashing in production

Reply:
Changed to DCHECK

Reply:
Cool thanks - yeah probably useful to use error logic as well if the several starts is a case we want to handle as well.